### PR TITLE
Fix/agent search form validation

### PR DESCRIPTION
**Pull request by VirginiaBalseiro**
This PR fixes:
- Uncaught promise if "Add Contact" button clicked without WebId
- "Insecure Resource Requested" when trying to add a WebId with `http`

## To test
- This component can be tested in Add Contact and in Permissions for an individual resource
- Verify that the form cannot be submitted without a valid url that doesn't start with `https`

- [ ] I've added a unit test to test for potential regressions of this bug.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: components/agentSearchForm/index.jsx [hunk 1]
```diff
@@ -22,34 +22,38 @@
...
   return (
-    <>
-      <Input id={inputId} label="WebID" onChange={onChange} value={agentId} />
+    <Form onSubmit={handleSubmit}>
+      <Input
+        id={inputId}
+        label="WebID"
+        onChange={handleChange}
+        value={agentId}
+        required
```
Comment (ajacksified) [inline components/agentSearchForm/index.jsx:47]: Arne noted in #100 that `required` gave the textbox a red border when you load the page- we may want to remove this field and check it in onSubmit instead.

Comment (megoth) [inline components/agentSearchForm/index.jsx:47]: This makes the field invalid from the getgo - Although not optimal, I think we should not use this and rather have a test in the submit function that checks if there's a value before submitting (i.e. [what I did in this PR](https://github.com/inrupt/pod-browser/pull/100/commits/d0aa1b6135fd3c9ce21bc1bd48e79e01329d5f89#diff-1f62fb90f81f3d9cdb374658a11dc25e234a621460729589a97797d315f1fcccR51-R53)).

**Review by megoth — CHANGES_REQUESTED** [review]

**Review by megoth — CHANGES_REQUESTED** [review]
Given the custom validation of empty strings there should also be a test that verifies that the form is not submitted for empty strings.

**Review by ajacksified — APPROVED** [review]
